linked-list/linked-list.spec.js

```diff
@@ -0,0 +1,72 @@
+import LinkedList from './linked-list';
+
+describe('LinkedList', () => {
+  it('push/pop', () => {
+    const list = new LinkedList();
+    list.push(10);
+    list.push(20);
+    expect(list.pop()).toBe(20);
+    expect(list.pop()).toBe(10);
+  });
+  xit('push/shift', () => {
+    const list = new LinkedList();
+    list.push(10);
+    list.push(20);
+    expect(list.shift()).toBe(10);
+    expect(list.shift()).toBe(20);
+  });
+  xit('unshift/shift', () => {
+    const list = new LinkedList();
+    list.unshift(10);
+    list.unshift(20);
+    expect(list.shift()).toBe(20);
+    expect(list.shift()).toBe(10);
+  });
+  xit('unshift/pop', () => {
+    const list = new LinkedList();
+    list.unshift(10);
+    list.unshift(20);
+    expect(list.pop()).toBe(10);
+    expect(list.pop()).toBe(20);
+  });
+  xit('example', () => {
+    const list = new LinkedList();
+    list.push(10);
+    list.push(20);
+    expect(list.pop()).toBe(20);
+    list.push(30);
+    expect(list.shift()).toBe(10);
+    list.unshift(40);
+    list.push(50);
+    expect(list.shift()).toBe(40);
+    expect(list.pop()).toBe(50);
+    expect(list.shift()).toBe(30);
+  });
+  xit('pops undefined when there are no elements', () => {
+    const list = new LinkedList();
+    expect(list.pop()).toBe(undefined);
+  });
+  xit('can count xits elements', () => {
+    const list = new LinkedList();
+    expect(list.count()).toBe(0);
+    list.push(10);
+    expect(list.count()).toBe(1);
+    list.push(20);
+    expect(list.count()).toBe(2);
+  });
+  xit('deletes the last element from the list', () => {
```

Comment:
The `delete` implementation deletes the element with the specified value, not the last element. The test description should reflect this.